### PR TITLE
[hotfix] DDF-4457-hotfix Fixes file system encryption issues introduced by #4502 (#4633)

### PR DESCRIPTION
**Pull request by bakejeyner**
- Fixes issues with encrypted input stream breaking downstream tests
- Fixes issues with writing encrypted resources

Uncomment this block and include PR # of change against 2.13.x where appropriate.
When doing so, it is reasonable to delete much of the rest of the PR description, leaving only:
- This uncommented block with link to 2.13.x PR
- Reviewers tagged
- Teams tagged
- Committers tagged
- The link to the "Notes on Review Process"

##### ABBREVIATED REVIEW BETWEEN 2.13.X AND MASTER IS IN EFFECT
Link to 2.13.x PR: #4607 
____

#### What does this PR do?
Fixes a bug introduced by #4502.

The file system encryption PR changed what was getting returned on ingest. Originally, the response would contain an input stream + size containing the plaintext content. However, due to the file system encryption changes, the response contains the **encrypted** input stream + size. This was breaking downstream tests that were expecting the plaintext input stream + size to be returned.

The PR also had an issue with writing encrypted resources. The chunk size was 128 and **every** write would result in 128 bytes being written, even if the original resource didn't have 128 bytes left.

#### Who is reviewing it? 
@ahoffer 
@blen-desta 
@brjeter 
@garrettfreibott 
@stustison 

#### Select relevant component teams: 
@codice/security 

#### How should this be tested?
Follow the steps on #4607.

#### What are the relevant tickets?
Fixes a bug introduced by #4502.

#### Checklist:
- [ ] Documentation Updated
- [ ] Update / Add Threat Dragon models
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

**Comment by bakejeyner** [comment]
build now

File: catalog/core/catalog-core-localstorageprovider/src/main/java/org/codice/ddf/catalog/content/impl/FileSystemStorageProvider.java [hunk 1]
```diff
@@ -639,17 +640,25 @@ private ContentItem generateContentFile(
...
       }
-      byteSource = com.google.common.io.Files.asByteSource(contentItemPath.toFile());
+      byteSource =
+          new ByteSource() {
+            @Override
+            public InputStream openStream() throws IOException {
+              InputStream fileInputStream = new FileInputStream(contentItemPath.toFile());
+              return crypter.decrypt(fileInputStream);
+            }
+          };
```
Comment (ahoffer) [inline catalog/core/catalog-core-localstorageprovider/src/main/java/org/codice/ddf/catalog/content/impl/FileSystemStorageProvider.java]: Clever to use anonymous classes.

**Comment by ahoffer** [comment]
The method `generateContentFile` has gotten long (maybe it always was). The nitty-gritty implementation details make it hard to understand what the method intends to accomplish and what paths it can take to reach its goal.
If this code is ever refactored, I recommend breaking it into smaller functions and using intentional-revealing function names (c.f. https://medium.freecodecamp.org/how-to-make-your-code-better-with-intention-revealing-function-names-6c8b5271693e) to improve readability.

**Comment by tyler30clemens** [comment]
Hero Successful ✅

had to manually change DDF version in Alliance to `2.15.0-SNAPSHOT`

**Comment by bakejeyner** [comment]
build now
(🤞 Looks like it failed on an unexpected AccessControlException)

**Comment by bakejeyner** [comment]
build now

**Comment by bakejeyner** [comment]
build now
(looks like there was an issue with Jenkins Spys 👀 )

**Comment by bakejeyner** [comment]
build now
(have a feeling CI failure is unrelated)

**Comment by bakejeyner** [comment]
build now
(CI failed for another, unassumingly unrelated, reason)

**Comment by blen-desta** [comment]
build now

**Comment by bakejeyner** [comment]
build now
(going to try one more time before I dive into whats going on)

**Comment by bakejeyner** [comment]
build now
(connectivity issues)

**Pull request merged**